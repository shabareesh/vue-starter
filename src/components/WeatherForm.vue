<template>
  <div class="form-wrapper">
    <h1>Add Weather</h1>
    <router-link
      :to="{name: 'Weather-list'}"
      class="nav-link"
      active-class="active">
      View list
    </router-link>
    <div class="form-container">
      <div class="form-field">
        <span>Wind Speed</span>
        <input type="number" />
      </div>
      <div class="form-field">
        <span>Current Speed</span>
        <input type="number" />
      </div>
      <div class="form-field">
        <span>Swell Hght</span>
        <input type="number" />
      </div>
      <div class="form-field">
        <span>Sea Hght</span>
        <input type="number" />
      </div>
    </div>
  </div>
</template>

<script>

export default {
  name: 'WeatherForm',
};
</script>
