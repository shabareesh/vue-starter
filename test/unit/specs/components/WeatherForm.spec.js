import vue from 'vue';
import { shallow } from 'vue-test-utils';
import { createRenderer } from 'vue-server-renderer';

import WeatherForm from '@/components/WeatherForm';

const weatherList = [
  { id: 1, windSpeed: '1', currentSpeed: '1', swellHght: '1', seaHght: '1', status: 'lock' },
  { id: 2, windSpeed: '2', currentSpeed: '2', swellHght: '2', seaHght: '2', status: 'lock' },
  { id: 4, windSpeed: '4', currentSpeed: '4', swellHght: '4', seaHght: '4', status: 'unlock' },
  { id: 5, windSpeed: '5', currentSpeed: '5', swellHght: '5', seaHght: '5', status: 'lock' },
];

describe('WeatherForm', () => {
  it('should render', () => {
    const renderer = createRenderer()
    const wrapper = shallow(WeatherForm, { data: { weatherList } });
    renderer.renderToString(wrapper.vm, (err, str) => {
      if (err) throw new Error(err);
      expect(str).toMatchSnapshot();
    });
  });
});
