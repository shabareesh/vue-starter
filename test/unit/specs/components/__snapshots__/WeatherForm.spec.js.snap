// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WeatherForm should render 1`] = `
<div class="form-wrapper">
    <h1>Add Weather</h1>
    <router-link to="[object Object]" active-class="active" class="nav-link">
        View list
    </router-link>
    <div class="form-container">
        <div class="form-field"><span>Wind Speed</span>
            <input type="number">
        </div>
        <div class="form-field"><span>Current Speed</span>
            <input type="number">
        </div>
        <div class="form-field"><span>Swell Hght</span>
            <input type="number">
        </div>
        <div class="form-field"><span>Sea Hght</span>
            <input type="number">
        </div>
    </div>
</div>
`;
